### PR TITLE
Minimal pre-commit config

### PR DESCRIPTION
**Pull request by Kilo59**
Add minimal CI checks such as linting and formatting

### CODE REVIEW

**Review by tyler-hoffman — COMMENTED** [review]
This looks correct, but is it complete?

* I would expect to see `pre-commit` in `requirements.txt`. Any reason not to do that?
* Probably worth mentioning running `pre-commit install` in the readme

I forget - how does github know to pick up the precommit config?

**Comment by Kilo59** [comment]
> This looks correct, but is it complete?

Installing pre-commit locally is not required for the CI to run and make changes on your behalf.

I was intentionally leaving that out so as not to add additional install steps to such an infrequently contributed to package. I can add if you still want but that was my reasoning.

The pre-commit ci github action is enabled for all great-expectations repos, but it won't do anything unless it sees a `.pre-commit-config.yaml` file.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# For available hooks and configuration options see: https://pre-commit.com/hooks.html
+# To manage and run locally, first install `pre-commit` https://pre-commit.com/#installation
+# To auto update to the latest versions `pre-commit autoupdate`
+# To manually run against all files `pre-commit run --all-files`
+# To force hooks to run on local commits `pre-commit install`
+repos:
```
Comment (Kilo59) [inline .pre-commit-config.yaml:6]: @tyler-hoffman added usage explanation here

**Review by tyler-hoffman — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (tyler-hoffman) [reply .pre-commit-config.yaml:7]: Thanks. I think this is great, and your reasoning about not updating the readme seems fine.

**Review by tyler-hoffman — APPROVED** [review]
Thanks!

**Pull request merged**